js/gestionPresupuesto.js

```diff
@@ -1,18 +1,163 @@
-// TODO: Crear las funciones, objetos y variables indicadas en el enunciado
+// Variables globales
+let presupuesto = 0;
+let gastos = [];
+let idGasto = 0;
 
-// TODO: Variable global
+// MIO
+function actualizarPresupuesto(num) {
+    if(num >= 0){
+        presupuesto = num;
+        return presupuesto;
+    }
+    else{
+        console.log("No se puede introducir un número negativo.");
+        return -1;
+    }
+}
 
 
-function actualizarPresupuesto() {
-    // TODO
+// MIO
+function mostrarPresupuesto() {
+    return `Tu presupuesto actual es de ${presupuesto} €`;
 }
 
-function mostrarPresupuesto() {
-    // TODO
+function CrearGasto(descripcion, valor, fecha, ...etiquetas) {
+
+    // Métodos
+    // MIO
+    this.mostrarGasto = function(){
+        return `Gasto correspondiente a ${this.descripcion} con valor ${this.valor} €`;
+    }
+
+    // MIO
+    this.actualizarDescripcion = function(descripcion){
+        this.descripcion = descripcion;
+        return descripcion;
+    }
+
+    // MIO
+    this.actualizarValor = function(valor){
+        if(valor >= 0){
+            this.valor = valor;
+            return valor;
+        }
+        else{
+            valor = this.valor;
+            return valor;
+        }
+    }
+
+    // MIO
+    this.mostrarGastoCompleto = function(){
+        let texto = `Gasto correspondiente a ${this.descripcion} con valor ${this.valor} €.
+Fecha: ${new Date(this.fecha).toLocaleString()}
+Etiquetas:\n`
+            for(let e of this.etiquetas){
+                texto += `- ${e}\n`
+            }
+            return texto;
+    }
+
+    this.actualizarFecha = function(fecha) {
+        let f = Date.parse(fecha);
+        if (f) {
+            this.fecha = f;
+        }
+    }
+
+    // MIO
+    this.anyadirEtiquetas = function(...etiqueta){
+        for(let e of etiqueta){
+            if(this.etiquetas.indexOf(e) == -1){
+                this.etiquetas.push(e);
+            }
+        }
+    }
+    // SOLUCION
+    this.borrarEtiquetas = function(...etqs) {
+        // No entiendo porqué se pone push si se supone que hay que borrar la etiqueta que
```

Comment:
La solución en este caso utiliza un nuevo array vacío, `newetiquetas`. Recorre las etiquetas existentes (`this.etiquetas`) y comprueba si cada una de ellas existe en el array de etiquetas a eliminar (`etqs`), mediante `indexOf`. Por último, solo añade la etiqueta existente que no forme parte del array de etiquetas a eliminar.

Espero habértelo aclarado... Si no es así, dímelo :wink: